app/controllers/users_controller.rb

```diff
@@ -27,19 +27,61 @@ def create
     end
   end
 
-  def update
+  def edit
+    @genres = Genre.all
+    @musician_skill = MusicianSkill.all
+    @user = current_user
+    @id = current_user.id
+    @image = current_user.image
   end
 
-  def destroy
-    if @user.destroy
-      head :no_content
+  def update
+    user = params[:user]
+    if user[:password] != user[:confirm_password]
+      flash[:alert] = 'confirm_password'
+      redirect_to action: :edit, id: current_user.id
     else
-      render json: @user.errors, status: :unprocessable_entity
+      update_fields(user)
+      redirect_to action: :show, id: current_user.id
     end
   end
 
   private
 
+  def update_fields(user)
+    create_photo(user) if user[:image]
+    update_user(user)
+    update_genres(user)
+    update_musician_skills(user)
+  end
+
+  def update_genres(user)
+    user[:genres].split(',').each do |item|
+      current_user.genres << Genre.find(item)
+    end
+  end
+
+  def update_musician_skills(user)
+    user[:musician_skills].split(',').each do |item|
+      current_user.musician_skills << MusicianSkill.find(item.to_i)
```

Comment:
`.to_i` is redundant. `.find` works fine with numeric strings

Reply:
like here
https://github.com/rubizza-camp/uber-music/pull/65/files#diff-4e05ad0d64e6100656b63ad1e78f32c5R60